client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-header.tsx

```diff
@@ -26,13 +28,25 @@ const ScreenHeader = ( { selectedPattern, onSelect, onBack, onDoneClick }: Props
 			<div className="screen-container__body">
 				<PatternSelector
 					patterns={ patterns }
-					onSelect={ onSelect }
+					onSelect={ ( selectedPattern ) => onSelect( 'header', selectedPattern, 'header' ) }
```

Comment:
I don't like to repeat the param `header` here, but one is the type and the other is the category. However, it is better than using `null` and then managing it in `onSelect`.